go/signatures.go

```diff
@@ -0,0 +1,92 @@
+package ct
+
+import (
+	"crypto"
+	"crypto/ecdsa"
+	"crypto/rsa"
+	"crypto/x509"
+	"encoding/asn1"
+	"encoding/pem"
+	"errors"
+	"fmt"
+	"log"
+	"math/big"
+)
+
+// PublicKeyFromPEM parses a PEM formatted block and returns the public key contained within.
+func PublicKeyFromPEM(b []byte) (crypto.PublicKey, error) {
+	p, _ := pem.Decode(b)
+	if p == nil {
+		return nil, fmt.Errorf("no PEM block found in %s", string(b))
+	}
+	return x509.ParsePKIXPublicKey(p.Bytes)
+}
+
+// SignatureVerifier can verify signatures on SCTs and STHs
+type SignatureVerifier struct {
+	pubKey crypto.PublicKey
+}
+
+// NewSignatureVerifier creates a new SignatureVerifier using the passed in PublicKey.
+func NewSignatureVerifier(pk crypto.PublicKey) SignatureVerifier {
```

Comment:
Should we verify key correctness here?

a) Known algorithm?

b) For RSA: big enough

c) For ECDSA: on the right curve?